src/rdpc/analysesProcessor.ts

```diff
@@ -206,13 +207,33 @@ export const getAllMergedDonor = async ({
         logger.info(`No ${analysisType} analyses for streaming`);
       }
       logger.info(`Streaming ${page.length} of ${analysisType} analyses...`);
-      const donorPerPage = toDonorCentric(page);
+      const filteredAnalyses = removeRunsWithSuppressedAnalyses(page);
+      const donorPerPage = toDonorCentric(filteredAnalyses);
       getAllRunsByAnalysesByDonors(mergedDonors, donorPerPage);
     }
   }
   return mergedDonors;
 };
 
+// Removes runs with suppressed producedAnalyses
+export const removeRunsWithSuppressedAnalyses = (
+  analyses: Analysis[]
+): Analysis[] => {
+  const result = analyses.reduce<Analysis[]>((acc, analysis) => {
+    const filteredRuns = analysis.runs.reduce<Run[]>((_acc, run) => {
```

Comment:
Don't be afraid to use longer names to make the different accumulators more readable. Maybe:
`resultAcc` and `runsAcc`

Reply:
This just looks like a filter operation, can you simplify this block?
```
const filteredRuns = analysis.runs.filter(run => run.producedAnalyses && run.producedAnalyses.length > 0);
```